### ISSUE TITLE
gaussian_reml_fit_latent does not optimize the latent coordinate (random init -> R2~0; only a near-correct init works; no recovered t in output)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#627]
## Summary

`gaussian_reml_fit_latent` is documented as estimating the per-row latent
coordinate `t_n` alongside `β` and `λ` ("each row has a latent coordinate
`t_n ∈ ℝ^d` … estimated per row alongside the spline coefficients β and the
smoothing parameters ρ"). In practice it does **not move the latent from its
init** — it just fits `β` at the positions you pass in. So the fit quality is
entirely determined by how good the *input* `t` already is, which defeats the
purpose of a latent-coordinate fit (you'd only have a good `t` if you'd already
solved the problem).

## Repro (gamfit 0.1.144)

Fit a shuffled parabola; vary only the init `t`:

```python
import numpy as np, torch, gamfit
torch.set_default_dtype(torch.float64)
r = np.random.default_rng(0); T = np.linspace(0,1,300,endpoint=False); u = 2*T-1
Y = np.c_[u, u**2-0.33]
perm = r.permutation(300); Yp = Y[perm]; true_t = T[perm]
M=14; centers=np.linspace(0,1,M)
Pn = np.asarray(gamfit.duchon_function_norm_penalty(torch.tensor(centers), m=2))
def fit(t0):
    o = gamfit.gaussian_reml_fit_latent(torch.tensor(np.asarray(t0,float).copy()),
            torch.tensor(Yp), len(Yp), 1, torch.tensor(centers), torch.tensor(Pn), m=2)
    f = np.asarray(o['fitted'])
    return 1 - ((Yp-f)**2).sum()/((Yp-Yp.mean(0))**2).sum()

print(fit(true_t))   # 1.0000  (correct order)
print(fit(r.random(300)))  # 0.0041  (random order — NOT optimized away)
```

```
init = true order      -> fit_R2 = 1.0000
init = random          -> fit_R2 = 0.0041   <-- t never optimized
init = reversed-true   -> fit_R2 = 1.0000   (reversal is a valid gauge, fine)
```

If the latent were actually estimated, all three inits would converge to the
same R² (the global optimum is init-independent). Instead the random init stays
at R²≈0 — the routine returns `β` for the *initial* `t` essentially unchanged.

## What I expected vs got
- Expected: an outer optimization over `t` (the doc mentions an IFT-warm-started
  outer optimizer and the `R_id(t,u)` identifiability regularizer), so a poor
  init is escaped.
- Got: a single inner solve of `β | t`; `t` is not advanced.

## Ask
Either (a) the low-level `gaussian_reml_fit_latent` should actually run the outer
latent optimization (return the converged `t`, which it currently doesn't expose
in the output dict either), or (b) document clearly that this low-level entry is
a single `β|t` solve and that the latent optimization lives only in the
high-level `LatentCoord` / `Model` path — and show a minimal end-to-end example
of fitting a `LatentCoord` (I couldn't find one; `smooth.py:859` constructs a
`LatentCoord` but never fits it, and `select_topology` needs the full formula
API). The output dict also has no `t`/`latent`/`coords` key, so even a successful
latent fit gives no way to read the recovered coordinate.

## Why it matters
This is the natural primitive for fitting a manifold feature with a *steerable
coordinate* (the position on the curve). As-is it can't discover the
parameterization — it can only confirm one you already have.

## Environment
gamfit 0.1.144, torch 2.12, CPU, macOS arm64.

**Comment by SauersML** [comment]
Confirming this still repros on released **gamfit 0.1.145**: with `gaussian_reml_fit_latent` on a 1D latent (latent_dim=1), true-init gives R²=1.000 but random-init gives R²=0.004 — t is never advanced, matching the root cause above (it's the inner β|t solve only).

Downstream impact, for prioritization: this is the one gamfit primitive that *should* directly solve unsupervised manifold-feature recovery (latent coordinate = the steering position on the curve). Because it doesn't optimize t, my Manifold-SAE work can't use it to discover positions — I'm forced to a geometric warm-start (PCA/angle init) + the closed-form `gaussian_reml_fit_positions` decoder instead. A working outer driver here (option-a plan) would let the position itself be REML-optimized, which is exactly what's needed under superposition where a good geometric init isn't available. No rush on top of the in-progress fix — just flagging it's a live blocker, not hypothetical.
